graph/ForceDirectedGraph/ForceDirectedGraph.tsx

```diff
@@ -0,0 +1,310 @@
+  const handlePanGestureEvent = (event: {
+    nativeEvent: { translationX: number; translationY: number };
+  }) => {
+    setTotalOffset({
+      x: event.nativeEvent.translationX / lastScale,
+      y: event.nativeEvent.translationY / lastScale,
+    });
+  };
+
+  // Handle Dragging State Change
+  const handlePanHandlerStateChange = (event: {
+    nativeEvent: { state: number };
+  }) => {
+    if (event.nativeEvent.state === State.END) {
+      setNodes(
+        nodes.map((node) => ({
+          ...node,
+          x: coordX(node),
+          y: coordY(node),
+        }))
+      );
+
+      if (!dragEnabled) {
+        setClickedNodeID(DEFAULT_CLICKED_NODE_ID);
+      }
+      setTotalOffset({ x: 0, y: 0 });
+    }
+  };
+
+  // Handle Zooming
+  const handlePinchGestureEvent = (event: {
+    nativeEvent: { scale: React.SetStateAction<number> };
+  }) => {
+    setScale(Number(event.nativeEvent.scale) * Number(lastScale));
+  };
+
+  const handlePinchHandlerStateChange = (event: {
+    nativeEvent: { state: number };
+  }) => {
+    if (event.nativeEvent.state === State.END) {
+      setLastScale(scale);
+    }
+  };
+
+  // Handle PRESS IN
+  const handlePressIn = (onPressCallback = () => {}) => {
+    onPressCallback();
+  };
+
+  // Handle PRESS OUT
+  const handlePressOut = (shortPressCallback = () => {}) => {
+    if (
+      Date.now() - pressStartRef.current < SHORT_PRESS_DURATION &&
+      Date.now() - pressStartRef.current > VERY_SHORT_PRESS_DURATION
+    ) {
+      shortPressCallback();
+    }
+  };
+
+  // Functions to get the X and Y coordinates of the nodes and the fill color of the nodes
+  const coordX = (node: Node): number => {
+    if (dragEnabled || clickedNodeID === node.id) {
+      return node.x + totalOffset.x;
+    }
+    return node.x;
+  };
+
+  const coordY = (node: Node): number => {
+    if (dragEnabled || clickedNodeID === node.id) {
+      return node.y + totalOffset.y;
+    }
+    return node.y;
+  };
+
+  // Create the lines and circles for the graph
+  const LINES = links.map((link) => (
+    <Line
+      key={link.source + link.target + "line"}
+      x1={coordX(
+        nodes.find((node) => node.id === link.source) ?? {
+          x: CENTER_WIDTH,
+          y: CENTER_HEIGHT,
+          id: link.source,
+          dx: 0,
+          dy: 0,
+        }
+      )}
+      y1={coordY(
+        nodes.find((node) => node.id === link.source) ?? {
+          x: CENTER_WIDTH,
+          y: CENTER_HEIGHT,
+          id: link.source,
+          dx: 0,
+          dy: 0,
+        }
+      )}
+      x2={coordX(
+        nodes.find((node) => node.id === link.target) ?? {
+          x: CENTER_WIDTH,
+          y: CENTER_HEIGHT,
+          id: link.target,
+          dx: 0,
+          dy: 0,
+        }
+      )}
+      y2={coordY(
+        nodes.find((node) => node.id === link.target) ?? {
+          x: CENTER_WIDTH,
+          y: CENTER_HEIGHT,
+          id: link.target,
+          dx: 0,
+          dy: 0,
```

Comment:
Might be nice to create a helper function to create objects of this type, maybe one accepting only `x`, y` and `id` and defaulting to zero whenever `dx` and `dy` are not specified.

Reply:
I was planning to create a specific Node Class instead of an interface to account for these kinds of methods during the next milestone (when the database would be linked).

So I think it's good for now.

Tell me what you think.